src/anonymizer.rs

```diff
@@ -0,0 +1,262 @@
+use std::collections::HashMap;
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+use std::num::Wrapping;
+
+pub trait AnonymizingClone {
+    fn masked_clone(&self, anon: &mut StatefulSdpAnonymizer) -> Self;
+}
+
+pub trait ToBytesVec {
+    fn to_byte_vec(&self) -> Vec<u8>;
+}
+
+impl ToBytesVec for u64 {
+    fn to_byte_vec(&self) -> Vec<u8> {
+        let mut bytes = Vec::new();
+        let mut val = *self;
+        for _ in 0..8 {
+            bytes.push(val as u8);
+            val <<= 8;
+        }
+        bytes.reverse();
```

Comment:
Can't you just omit the semicolon on this line and delete the next line?

Reply:
reverse() doesn't return Self. Oh, for Ruby's `tap`.